packages/beacon-node/src/sync/backfill/backfill.ts

```diff
@@ -866,7 +866,7 @@ async function extractPreviousFinOrWsCheckpoint(
   )[0];
 
   let prevFinalizedCheckpointBlock: BackfillBlockHeader;
-  if (nextPrevFinOrWsBlock != null) {
+  if (nextPrevFinOrWsBlock !== undefined) {
```

Comment:
type of `nextPrevFinOrWsBlock` is ` allForks.SignedBeaconBlock` compiler ensures can't be null. The actual value is access via array index. If at runtime, nothing is found at that index, `undefined` will be set not `null`. So safe to do strict check against `undefined`